src/patchwork/annotations.ts

```diff
@@ -0,0 +1,406 @@
+          overlappingAnnotations
+        ),
+        discussion,
+      });
+    });
+
+    const computedAnnotationGroups: AnnotationGroup<unknown, unknown>[] =
+      groupAnnotations(
+        docType,
+        editAnnotations.filter(
+          (annotation) => !claimedAnnotations.has(annotation)
+        )
+      ).map((annotations) => ({ annotations }));
+
+    const combinedAnnotationGroups = discussionGroups.concat(
+      computedAnnotationGroups
+    );
+
+    const sortAnchorsBy = docTypes[docType].sortAnchorsBy;
+
+    return {
+      annotations: editAnnotations.concat(highlightAnnotations),
+      annotationGroups: sortAnchorsBy
+        ? sortBy(combinedAnnotationGroups, (annotationGroup) =>
+            min(
+              annotationGroup.annotations.map((annotation) =>
+                sortAnchorsBy(doc, annotation.target)
+              )
+            )
+          )
+        : combinedAnnotationGroups,
+    };
+  }, [doc, discussions, diff]);
+
+  const {
+    selectedAnchors,
+    hoveredAnchors,
+    selectedAnnotationGroupIds,
+    expandedAnnotationGroupId,
+  } = useMemo(() => {
+    const selectedAnchors = new Set<unknown>();
+    const hoveredAnchors = new Set<unknown>();
+    const selectedAnnotationGroupIds = new Set<string>();
+    let expandedAnnotationGroupId: string;
+
+    // Record selection state for anchors and annotation groups
+    switch (selectedState?.type) {
+      case "anchors": {
+        // focus selected anchors
+        selectedState.anchors.forEach((anchor) => selectedAnchors.add(anchor));
+
+        // first annotationGroup that contains all selected anchors is expanded
+        const annotationGroup = annotationGroups.find((group) =>
+          doesAnnotationGroupContainAnchors(
+            docType,
+            group,
+            selectedState.anchors,
+            doc
+          )
+        );
+        if (annotationGroup) {
+          expandedAnnotationGroupId = getAnnotationGroupId(annotationGroup);
+
+          // ... the anchors in that group are focused as well
+          annotationGroup.annotations.map((annotation) =>
+            selectedAnchors.add(annotation.target)
+          );
+        }
+        break;
+      }
+
+      case "annotationGroup": {
+        const annotationGroup = annotationGroups.find(
+          (group) => getAnnotationGroupId(group) === selectedState.id
+        );
+
+        if (annotationGroup) {
+          // expand seleted annotation group
+          expandedAnnotationGroupId = selectedState.id;
+
+          // focus all anchors in the annotation group
+          annotationGroup.annotations.forEach((annotation) =>
+            selectedAnchors.add(annotation.target)
+          );
+        }
+        break;
+      }
+    }
+
+    // Record hovered state for anchors
+    switch (hoveredState?.type) {
+      case "anchor": {
+        // focus hovered anchor
+        hoveredAnchors.add(hoveredState.anchor);
```

Comment:
I think you changed the behavior that on hover it only highlights the anchor itself. I'm curious why you like this more. I thought it was nice that on hover in the sidebar, would highlight the discussions relating to it.